src/compiler/utilities.ts

```diff
@@ -364,6 +364,23 @@ namespace ts {
         return createTextSpanFromBounds(start, scanner.getTextPos());
     }
 
+    function getErrorSpanForArrowFunction(sourceFile: SourceFile, node: ArrowFunction): TextSpan {
+        const pos = skipTrivia(sourceFile.text, node.pos);
+
+        if (node.body && node.body.kind === SyntaxKind.Block) {
+            const {line: startLine } = getLineAndCharacterOfPosition(sourceFile, node.body.pos);
+            const {line: endLine } = getLineAndCharacterOfPosition(sourceFile, node.body.end);
+            if (startLine < endLine) {
+                // The arrow function body spans multiple lines, 
+                // make the error span be the first line.
+                const endOfFirstLine = getLineStarts(sourceFile)[startLine + 1] - 1;

```

Comment:
I don't get why you need to do -1 here? shouldn't the etLineStarts(sourceFile)[startLine + 1] be enough?